contrib/googlejavaformat/tests/python/pants_test/contrib/googlejavaformat/test_googlejavaformat.py

```diff
@@ -5,45 +5,114 @@
-    self.assert_failure(failing_test)
-
-  def test_disabled(self):
-    target = '{}/badgooglejavaformat'.format(TEST_DIR)
-    # test should pass because check is disabled.
-    failing_test = self.run_pants(['lint', target],
-      {'lint.google-java-format': {'skip':'True'}})
-    self.assert_success(failing_test)
-
-  def format_file_and_verify_fmt(self, options):
-    # take a snapshot of the file which we can write out
-    # after the test finishes executing.
-    test_file_name = '{}/badscalastyle/BadGoogleJavaFormat.java'.format(TEST_DIR)
-    with open(test_file_name, 'r') as f:
-      contents = f.read()
-
-    try:
-      # format an incorrectly formatted file.
-      target = '{}/badgooglejavaformat'.format(TEST_DIR)
-      fmt_result = self.run_pants(['fmt', target], {'fmt.google-java-format':options})
-      self.assert_success(fmt_result)
-
-      # verify that the lint check passes.
-      test_fmt = self.run_pants(['lint', target], {'lint.google-java-format':options})
-      self.assert_success(test_fmt)
-    finally:
-      # restore the file to its original state.
-      f = open(test_file_name, 'w')
-      f.write(contents)
-      f.close()
+from textwrap import dedent
+
+from pants.backend.jvm.register import build_file_aliases as register_jvm
+from pants.backend.jvm.targets.java_library import JavaLibrary
+from pants.base.exceptions import TaskError
+from pants.build_graph.register import build_file_aliases as register_core
+from pants_test.jvm.nailgun_task_test_base import NailgunTaskTestBase
+
+from pants.contrib.googlejavaformat.googlejavaformat import (GoogleJavaFormat,
+                                                             GoogleJavaFormatCheckFormat)
+
+
+class BaseTest(NailgunTaskTestBase):
+
+  _BADFORMAT = dedent("""
+    package org.pantsbuild.contrib.googlejavaformat;
+    class MyClass {
+        public static void main(String[] args) {
+            System.out.println("Hello google-java-format!");
+        }
+    }
+    """)
+
+  _GOODFORMAT = dedent("""\
+    package org.pantsbuild.contrib.googlejavaformat;
+    
+    class MyClass {
+      public static void main(String[] args) {
+        System.out.println("Hello google-java-format!");
+      }
+    }
+  """)
+
+
+class GoogleJavaFormatTests(BaseTest):
+
+  @classmethod
+  def task_type(cls):
+    return GoogleJavaFormat
+
+  @property
+  def alias_groups(self):
+    return super(GoogleJavaFormatTests, self).alias_groups.merge(
+      register_core().merge(register_jvm())
+    )
+
+  def test_googlejavaformat(self):
+    javafile = self.create_file(
+      relpath='src/java/org/pantsbuild/contrib/googlejavaformat/MyClass.java',
+      contents=self._BADFORMAT)
+    target = self.make_target(
+      spec='src/java/org/pantsbuild/contrib/googlejavaformat',
+      target_type=JavaLibrary,
+      sources=['MyClass.java'],
+    )
+
+    context = self.context(target_roots=[target])
+    task = self.prepare_execute(context)
```

Comment:
You could just `self.execute(context)` since you don't need the task object returned by `prepare_execute`. Same goes in a few places below.

Reply:
This took some playing around with earlier to get working - without `prepare_execute` I was running into issues about `product`'s needing to be registered or something like that. I'll take another look at this.

Reply:
Must have been related to something else - fixed! Now that I think about it, I am probably mixing this up with an error from before I switched to subclassing `NailgunTaskTestBase`.